test/mocks/upstream/client.go

```diff
@@ -0,0 +1,202 @@
+package upstream
```

Comment:
Rather than putting this in a giant mock repo, I think it might be better to put this in the same repo as upstream package, like so:

`internal/app/upstream/mock.go`
or
`internal/app/upstream/mock/client.go`

Modules using this mock will still need to import `"github.com/envoyproxy/xds-relay/internal/app/upstream"` anyway to use it. Might as well put them together.

Reply:
Sure can do that, i think i moved this to mock package out so that other packages can use based on a previous comment. I like how explicitly other packages can import mock [here](https://github.com/envoyproxy/xds-relay/blob/0ac6a265387b46e81d5f10805ef06e1f8398b82d/internal/app/orchestrator/orchestrator_test.go#L11) [here](https://github.com/envoyproxy/xds-relay/blob/0ac6a265387b46e81d5f10805ef06e1f8398b82d/internal/app/upstream/client_test.go#L13).
LMK if you feel strongly about moving the file into the paths you suggested.

Reply:
You can achieve the same result by putting the mocks in the `/mock` folder.

Putting the mock with the struct it is mocking makes it so that the dependencies are clear. If you are not careful, you can also end up with circular dependencies by putting them in one module.

Reply:
sounds good. I've moved the mocks into `internal/app/upstream/mock`.